gix-fs/tests/stack/mod.rs

```diff
@@ -26,6 +27,287 @@ impl gix_fs::stack::Delegate for Record {
+    assert!(
+        absolute.is_relative(),
+        "on Windows, strange single-backslash paths are relative (and relative to the current drive)"
+    );
+    assert_eq!(
+        p("relative").join(absolute),
+        absolute,
+        "relative + absolute = absolute - however, they kind of act like they are absolute in conjunction with relative base paths"
+    );
+    assert_eq!(
+        p("relative").join(bs_absolute),
+        bs_absolute,
+        "relative + absolute = absolute - backslashes aren't special here, and it just acts like it's absolute"
+    );
+
+    assert_eq!(
+        p("c:").join("relative"),
+        p("c:relative"),
+        "drive + relative = strange joined result with missing backslash, but it's a valid path that works just like `c:\relative`"
+    );
+    assert_eq!(
+        p("c:\\").join("relative"),
+        p("c:\\relative"),
+        "absolute + relative = joined result"
+    );
+
+    assert_eq!(
+        p("\\\\?\\base").join(absolute),
+        p("\\\\?\\base\\absolute"),
+        "absolute1 + unix-absolute2 = joined result with backslash"
+    );
+    assert_eq!(
+        p("\\\\.\\base").join(absolute),
+        p("\\\\.\\base\\absolute"),
+        "absolute1 + absolute2 = joined result with backslash (device namespace)"
+    );
+    assert_eq!(
+        p("\\\\?\\base").join(bs_absolute),
+        p("\\\\?\\base\\absolute"),
+        "absolute1 + absolute2 = joined result"
+    );
+    assert_eq!(
+        p("\\\\.\\base").join(bs_absolute),
+        p("\\\\.\\base\\absolute"),
+        "absolute1 + absolute2 = joined result (device namespace)"
+    );
+
+    assert_eq!(p("/").join("C:"), p("C:"), "unix-absolute + win-drive = win-drive");
+    assert_eq!(
+        p("d:/").join("C:"),
+        p("C:"),
+        "d-drive + c-drive = c-drive - interesting, as C: is supposed to be relative"
+    );
+    assert_eq!(
+        p("d:\\").join("C:\\"),
+        p("C:\\"),
+        "d-drive-with-bs + c-drive-with-bs = c-drive-with-bs - nothing special happens with backslashes"
+    );
+    assert_eq!(
+        p("c:\\").join("\\\\.\\"),
+        p("\\\\.\\"),
+        "c-drive-with-bs + device-namespace-unc = device-namespace-unc"
+    );
+    assert_eq!(
+        p("/").join("C:/"),
+        p("C:\\"),
+        "unix-absolute + win-drive = win-drive, strangely enough it changed the trailing slash to backslash, so better not have trailing slashes"
+    );
+    assert_eq!(p("/").join("C:\\"), p("C:\\"), "unix-absolute + win-drive = win-drive");
+    assert_eq!(
+        p("\\\\.").join("C:"),
+        p("C:"),
+        "device-namespace-unc + win-drive-relative = win-drive-relative - c: was supposed to be relative, but it's not acting like it."
+    );
+    assert_eq!(p("relative").join("C:"), p("C:"), "relative + win-drive = win-drive");
+
+    assert_eq!(
+        p("/").join("\\\\localhost"),
+        p("\\localhost"),
+        "unix-absolute + win-absolute-unc = win-absolute-unc"
```

Comment:
These are not raw strings, so this is indicating that `/` joined with `\\localhost` gives `\localhost` with one leading `\`, and I don't understand why this is what happens. It also does not appear to be what this meant to assert, since it characterizes the effect of the join as `= win-absolute-unc`, but a path that begins with only one `\` is not a UNC path. So I don't know what's going on here.

Reply:
That's a mistake, as it indeed doesn't consider that "\\…" is not a raw string.
I don't know what an alternative comment would be, but it definitely isn't `= win-absolute-unc`.